README.md

```diff
@@ -10,15 +10,37 @@ Sporkfish is a Python-based chess engine. Chess programming techniques, although
 
 ### Set up
 
-To quickly set up the environment, you can use docker. From a terminal in the root directory:
+As a general case `Using DevContainer with VSCode` or `Using Github Codespace`
 
-```
-docker pull kylchiu/sporkfish-dev:latest
-docker build -t kylchiu/sporkfish-dev:latest .
-docker run -it kylchiu/sporkfish-dev:latest
-```
+Install: 
+- [VSCode](https://code.visualstudio.com/download)
+- [Docker Desktop](https://www.docker.com/products/docker-desktop/)
+
+#### Using DevContainer with VSCode 
+
+1. Make sure Docker Desktop is up and running.
+2. Open up the Command Palette and run `Dev Containers: Rebuild Container` 
+3. A warning might pop up saying `Not all host requirements in devcontainer.json are met by the Docker daemon.`, disregard this and continue. 
+4. The window should reload and you will see [Dev Container] in the URL bar as well as on the status bar bottom right of the window to indicate your setup is complete. 
+
+This will setup a Docker container to run in Docker Desktop with all the neceessary dependencies (python, pip, git, ...) with the correct versions. 
+
+\* Please exit out of the container to commit your git changes for now as the ssh keys aare not being mounted to the container. This will be fixed soon. 
+
+#### Using Github Codespace
+
+You do not need to install Docker Desktop with this option. 
+
+1. In this Github repository, click on the Code dropdown, select Codespace and click `Create`. 
+2. Continue through all prompts using default options. 
+
+This will setup a Docker container to run in Github instead of Docker Desktop. All else being equal, but an active internet connection will be required for this, and this will use up your monthly allowance for Github codespace. 
+
+#### Using DevContainer without VSCode
+
+You do not need to install VSCode with this option.
 
-This generates an interactive bash shell for you to run the program in.
+#TODO
```

Comment:
Could we keep the original run method as well pls? I'm aware of people who might not be using vscode